toxcore/group_chats.c

```diff
@@ -0,0 +1,7668 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2016-2020 The TokTok team.
+ * Copyright © 2015 Tox project.
+ */
+
+/*
+ * An implementation of massive text only group chats.
+ */
+
+#include <string.h>
+
+#include "DHT.h"
+#include "LAN_discovery.h"
+#include "Messenger.h"
+#include "TCP_connection.h"
+#include "friend_connection.h"
+#include "group_announce.h"
+#include "group_chats.h"
```

Comment:
The .h file for the current .c file should come at the top, before all other includes.